File: 35. Search Insert Position/35. Search Insert Position.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer}
def search_insert(nums, target)
    left = 0
    right = nums.length - 1
    
    while left <= right
        mid = left + (right - left) / 2
        
        if nums[mid] == target
            return mid
        elsif nums[mid] < target
            left = mid + 1
        else
            right = mid - 1
        end
    end
    
    left
end

File: 35. Search Insert Position/35. Search Insert Position.ts
function searchInsert(nums: number[], target: number): number {
    let left: number = 0;
    let right: number = nums.length - 1;
    
    while (left <= right) {
        const mid: number = left + Math.floor((right - left) / 2);
        
        if (nums[mid] === target) {
            return mid;
        } else if (nums[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    
    return left;
}